{{cookiecutter.project_slug}}/gulpfile.js

```diff
@@ -67,72 +74,80 @@ gulp.task('styles', function() {
     .pipe(rename({ suffix: '.min' }))
-    .pipe(cssnano()) // Minifies the result
-    .pipe(gulp.dest(paths.css));
-});
+    .pipe(postcss(minifyCss)) // Minifies the result
+    .pipe(dest(paths.css))
+}
 
 // Javascript minification
-gulp.task('scripts', function() {
-  return gulp.src(paths.js + '/project.js')
+function scripts() {
+  return src(paths.js + '/project.js')
     .pipe(plumber()) // Checks for errors
     .pipe(uglify()) // Minifies the js
     .pipe(rename({ suffix: '.min' }))
-    .pipe(gulp.dest(paths.js));
-});
-
+    .pipe(dest(paths.js))
+}
 
 {% if cookiecutter.custom_bootstrap_compilation == 'y' %}
 // Vendor Javascript minification
-gulp.task('vendor-scripts', function() {
-  return gulp.src(paths.vendorsJs)
+function vendorScripts() {
+  return src(paths.vendorsJs)
     .pipe(concat('vendors.js'))
-    .pipe(gulp.dest(paths.js))
+    .pipe(dest(paths.js))
     .pipe(plumber()) // Checks for errors
     .pipe(uglify()) // Minifies the js
     .pipe(rename({ suffix: '.min' }))
-    .pipe(gulp.dest(paths.js));
-});
+    .pipe(dest(paths.js))
+}
 {% endif %}
 
 // Image compression
-gulp.task('imgCompression', function(){
-  return gulp.src(paths.images + '/*')
+function imgCompression() {
+  return src(paths.images + '/*')
     .pipe(imagemin()) // Compresses PNG, JPEG, GIF and SVG images
-    .pipe(gulp.dest(paths.images))
-});
+    .pipe(dest(paths.images))
+}
+
+// Generate all assets
+const generateAssets = parallel(
+  styles,
+  scripts,
+  {% if cookiecutter.custom_bootstrap_compilation == 'y' %}vendorScripts,{% endif %}
+  imgCompression
+)
 
 // Run django server
-gulp.task('runServer', function(cb) {
-  var cmd = spawn('python', ['manage.py', 'runserver'], {stdio: 'inherit'});
+function runServer(cb) {
+  var cmd = spawn('python', ['manage.py', 'runserver'], {stdio: 'inherit'})
   cmd.on('close', function(code) {
-    console.log('runServer exited with code ' + code);
-    cb(code);
-  });
-});
+    console.log('runServer exited with code ' + code)
+    cb(code)
+  })
+}
 
 // Browser sync server for live reload
-gulp.task('browserSync', function() {
+function initBrowserSync() {
     browserSync.init(
       [paths.css + "/*.css", paths.js + "*.js", paths.templates + '*.html'], {
         proxy:  "localhost:8000"
-    });
-});
+    })
+}
 
 // Watch
-gulp.task('watch', function() {
-
-  gulp.watch(paths.sass + '/*.scss', ['styles']);
-  gulp.watch(paths.js + '/*.js', ['scripts']).on("change", reload);
-  gulp.watch(paths.images + '/*', ['imgCompression']);
-  gulp.watch(paths.templates + '/**/*.html').on("change", reload);
-
-});
-
-// Default task
-gulp.task('default', function() {
-    runSequence(['styles', 'scripts', {% if cookiecutter.custom_bootstrap_compilation == 'y' %}'vendor-scripts', {% endif %}'imgCompression'], ['runServer', 'browserSync', 'watch']);
-});
+function watchPaths() {
+  watch(paths.sass + '/*.scss', styles)
+  watch(paths.js + '/*.js', scripts).on("change", reload)
```

Comment:
I'm not sure if this was a problem until now, but it looks like it's also matching the minified version, and therefore it's stuck in a loop when the `project.js` is edited:

```
[15:56:40] Finished 'scripts' after 6.81 ms
[15:56:40] Starting 'scripts'...
[15:56:40] Finished 'scripts' after 5.94 ms
[15:56:40] Starting 'scripts'...
[15:56:40] Finished 'scripts' after 6.21 ms
[15:56:41] Starting 'scripts'...
[15:56:41] Finished 'scripts' after 5.34 ms
[15:56:41] Starting 'scripts'...
[15:56:41] Finished 'scripts' after 13 ms
[15:56:41] Starting 'scripts'...
[15:56:41] Finished 'scripts' after 6.05 ms
[15:56:42] Starting 'scripts'...
[15:56:42] Finished 'scripts' after 5.44 ms
[15:56:42] Starting 'scripts'...
[15:56:42] Finished 'scripts' after 5.5 ms
[15:56:42] Starting 'scripts'...
[15:56:42] Finished 'scripts' after 5.86 ms
[15:56:43] Starting 'scripts'...
[15:56:43] Finished 'scripts' after 5.96 ms
[15:56:43] Starting 'scripts'...
[15:56:43] Finished 'scripts' after 8.68 ms
[15:56:43] Starting 'scripts'...
[15:56:43] Finished 'scripts' after 5.88 ms
```

1. Source is updated
2. Triggers watch task
3. Regenerate minified version
4. Triggers watch task
5. Regenerate minified version
6. etc...

The first argument should be changed to a list of patterns with an exclusion pattern (starts with '!') as the second one. Since we are using modern JS in this file, we can also use templated strings. Putting all together:

```suggestion
  watch([`${paths.js}/*.js`,`!${paths.js}/*.min.js`], scripts ).on("change", reload)
```